src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -0,0 +1,113 @@
+import './NewCommentForm.scss';
+import React, { FormEvent, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Comment } from '../../react-app-env';
+import { postComment, getComments } from '../../api/comments';
+import { getCommentsSelector } from '../../store/selectors';
+import { setCommentsAction } from '../../store';
+
+export const NewCommentForm: React.FC = () => {
+  const dispatch = useDispatch();
+  const commentsList = useSelector(getCommentsSelector);
+  const [yourname, setYourname] = useState('');
+  const [youremail, setYouremail] = useState('');
+  const [yourcomment, setYourcomment] = useState('');
+  const [error, setError] = useState(false);
+
+  const addComment = async (newComment: Comment) => {
+    if (commentsList) {
+      postComment(
+        newComment.name,
+        newComment.email,
+        newComment.body,
+        newComment.postId,
+      );
+
+      const afterAdded = await getComments(newComment.postId);
```

Comment:
you throw an error in  getComments function so ypu need to catch error here, because afterAdded can be undefined if getComments will return an error
the same for another functions with fetch